### PR TITLE
Add an option to control on applying tensor.pad -> insert_slice pattern.

### PR DESCRIPTION
**Pull request by hanhanW**
This avoids having dispatches that only have single tensor.pad op. The tensor.pad dispatches are not supported. It makes the option more robust, and can be applied on all the tracking models.

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/PadTensorToTensorInsertSlice.cpp [hunk 1]
```diff
@@ -36,6 +36,9 @@ namespace {
 /// consumers.
 struct PadTensorOpConversion : public OpRewritePattern<tensor::PadOp> {
   using OpRewritePattern<tensor::PadOp>::OpRewritePattern;
+  PadTensorOpConversion(MLIRContext *context, bool skipOneLinalgUseCase)
+      : OpRewritePattern<tensor::PadOp>(context, skipOneLinalgUseCase),
+        skipOneLinalgUseCase(skipOneLinalgUseCase) {}
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/PadTensorToTensorInsertSlice.cpp:41]: Nit: `s/skipOneLinalgUseCase/skipSingleLinalgOpUses`

**Review by MaheshRavishankar — APPROVED** [review]
This entire pass needs to be deprecated anyway :P

**Review by antiagainst — APPROVED** [review]

**Comment by antiagainst** [comment]
Might be good to rename the pass properly given that `linalg.pad_tensor` was changed to `tensor.pad` long time ago. :)

**Comment by hanhanW** [comment]
> Might be good to rename the pass properly given that `linalg.pad_tensor` was changed to `tensor.pad` long time ago. :)

yes, I plan to send out a follow up for it!

**Pull request merged**